### PR TITLE
Add MNIST benchmark

### PR DESCRIPTION
**Pull request by apertovs**
```
****************************************************************************************
*                                     Dataset info                                     *
****************************************************************************************
- Name: MNIST
- Train size: 60000
- Validation size: 10000
- Image shape: 28x28
****************************************************************************************
*                                        Torch                                        *
****************************************************************************************
- Epoch:          1/10 | Train loss: 0.354176   | Validation accuracy: 0.9479    
- Epoch:          2/10 | Train loss: 0.178707   | Validation accuracy: 0.9631    
- Epoch:          3/10 | Train loss: 0.133331   | Validation accuracy: 0.9693    
- Epoch:          4/10 | Train loss: 0.109757   | Validation accuracy: 0.9692    
- Epoch:          5/10 | Train loss: 0.0950253  | Validation accuracy: 0.975     
- Epoch:          6/10 | Train loss: 0.0824357  | Validation accuracy: 0.9736    
- Epoch:          7/10 | Train loss: 0.0731623  | Validation accuracy: 0.9751    
- Epoch:          8/10 | Train loss: 0.0673887  | Validation accuracy: 0.9753    
- Epoch:          9/10 | Train loss: 0.0622156  | Validation accuracy: 0.9764    
- Epoch:         10/10 | Train loss: 0.0569681  | Validation accuracy: 0.975     
Train time: 17491 Inference time: 1897
****************************************************************************************
*                                      Deepworks                                      *
****************************************************************************************
- Epoch:          1/10 | Train loss: 0.371643   | Validation accuracy: 0.94401   
- Epoch:          2/10 | Train loss: 0.166691   | Validation accuracy: 0.959235  
- Epoch:          3/10 | Train loss: 0.125596   | Validation accuracy: 0.966747  
- Epoch:          4/10 | Train loss: 0.103472   | Validation accuracy: 0.969351  
- Epoch:          5/10 | Train loss: 0.086196   | Validation accuracy: 0.971154  
- Epoch:          6/10 | Train loss: 0.0767367  | Validation accuracy: 0.970653  
- Epoch:          7/10 | Train loss: 0.0681201  | Validation accuracy: 0.971454  
- Epoch:          8/10 | Train loss: 0.0617197  | Validation accuracy: 0.970152  
- Epoch:          9/10 | Train loss: 0.0543548  | Validation accuracy: 0.974259  
- Epoch:         10/10 | Train loss: 0.0517817  | Validation accuracy: 0.976162  
Train time: 20211 Inference time: 1084

========================================================================================
||                    ||     Deepworks      ||       Torch        ||    DW vs Torch     ||
========================================================================================
||       Epochs       ||         10         ||         10         ||        None        ||
||     Train time     ||      17491 ms      ||      20211 ms      ||      -15.55%       ||
||   Inference time   ||      1897 ms       ||      1084 ms       ||      +75.00%       ||
||      Accuracy      ||      0.975000      ||      0.976162      ||       -0.12%       ||
||        Loss        ||      0.056968      ||      0.051782      ||      +10.02%       ||
========================================================================================
```

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -19,9 +19,11 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 ########################################
 # Define basic flags
 ########################################
-option(BUILD_TESTS   "Build deepworks with tests"         ON)
-option(WITH_EIGEN    "Build deepworks with eigen backend" ON)
-option(BUILD_SAMPLES "Build samples"                      ON)
+option(BUILD_TESTS     "Build deepworks with tests"         ON)
+option(WITH_EIGEN      "Build deepworks with eigen backend" ON)
+option(BUILD_SAMPLES   "Build samples"                      ON)
+option(BUILD_BENCHMARK "Build benchmark"                    ON)
```
Comment (TolyaTalamanov) [inline CMakeLists.txt:25]: I hope it will `benchmark` - > `benchmarks`

File: CMakeLists.txt [hunk 2]
```diff
@@ -32,3 +34,7 @@ add_subdirectory(tests)
 if (BUILD_SAMPLES)
     add_subdirectory(samples)
 endif()
+
+if (BUILD_BENCHMARK)
```
Comment (TolyaTalamanov) [inline CMakeLists.txt:38]: the same and all the above

**Review by TolyaTalamanov — APPROVED** [review]
:fire: Great!

**Comment by TolyaTalamanov** [comment]
BTW Are you gong to add README.md ?

**Comment by TolyaTalamanov** [comment]
In fact I can merge it right now, because there is only small cosmetic fix required, you can fix it a bit later.

File: benchmark/mnist/benchmark_dw.hpp [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+private:
+    std::vector<DataInfo> m_info;
+};
+
+BenchmarkResults executeDeepworksMNISTBenchmark(dw::DataLoader& train_loader,
+                                                dw::DataLoader& val_loader,
+                                                size_t epochs) {
+    // Define model
+    auto model = buildMNISTModel(32); // TODO: set in runtime
```
Comment (TolyaTalamanov) [inline benchmark/mnist/benchmark_dw.hpp:59]: Oh, let's to provide an opportunity to set batch size via cmd arg

**Review by TolyaTalamanov — APPROVED** [review]
Provide and example how to run it in README.md, please

File: CMakeLists.txt (hunk 1 above)
Comment (TolyaTalamanov) [inline CMakeLists.txt:25]: It is still `benchmark`, isn't it ?

Comment (apertovs) [reply CMakeLists.txt]: Done.

**Comment by TolyaTalamanov** [comment]
Great, thanks! :+1:

**Pull request merged**